py/dml/types.py

```diff
@@ -386,16 +392,17 @@ def bytes(self):
     def sizeof(self):
         return self.bytes
 
-    def member_type_bits(self, member):
-        "Return the type, msb, and lsb for a member"
+    @property
+    def members_qualified(self):
         assert(self.is_bitfields)
-        for (name, typ, msb, lsb) in self.members:
-            if name == member:
-                if self.const and not typ.const:
-                    typ = typ.clone()
-                    typ.const = True
-                return typ, msb, lsb
-        return None, None, None
+        return ((name, conv_const(self.const, typ), msb, lsb)
+                for (name, (typ, msb, lsb)) in self.members.items())
+
+    def get_member_qualified(self, member):
+        t = self.members.get(member)
+        if t is not None:
+            t = (conv_const(self.const, t[0]),) + t[1:]
```

Comment:
nitpick: consider avoiding local mutation here, e.g. `return None if t is None else ...`

Reply:
Pssh, ternary operators? How about _boolean operators_?:
```
return t and (conv_const(self.const, t[0]),) + t[1:]
```

Reply:
Depends on how we think about non-None false values.

Reply:
I'm fine with it, as we know they can't occur in this context.